### ISSUE TITLE
Update to Opp.Account or Opp.PrimaryContact should trigger NPSP Opp rename

### ISSUE THREAD

**Issue by johnfine** [SalesforceFoundation/Cumulus#1876]
Issue: a Contact Merge (or Single-table dedup from DemandTools) will update Opp.Account and Opp.PrimaryContact.  If either the new Account name or PrimaryContact name are different than the old one, then Opp.Name may no longer reflect these new values (depending on which fields are chosen in NPSP Opp naming).

Proposed change: If NPSP Opp Naming is turned on, then these updates should cause Opp renaming for the affected Opp.

**Comment by judisohn** [comment]
I believe this is covered in #1360 - while that one is about custom fields, it's the same core issue where changes to fields used in custom naming are not triggering the update of the name.

**Comment by njjc** [comment]
That one is about household naming, we can leave this one open for opp naming.

**Comment by judisohn** [comment]
Good catch. Thanks, Nic.



On Dec 17, 2015, at 7:48 PM, Nicolas Campbell <notifications@github.com>
wrote:

That one is about household naming, we can leave this one open for opp
naming.

—
Reply to this email directly or view it on GitHub
<https://github.com/SalesforceFoundation/Cumulus/issues/1876#issuecomment-165626400>
.

**Comment by cdcarter** [comment]
Right now, it looks like Opp renaming is only being triggered at the before insert point, yes? Does this issue really expand to include renaming when name related fields are edited (standard (and custom))? @mpusto

**Comment by judisohn** [comment]
@cdcarter Yes, that's what I understand. And I think it may get a bit more complex than the similar issue on Household names because Household names are only pulled from their related contacts, while Opportunity Names can use fields from any object where there's a relationship field on Opportunity. Recurring Donation, Primary Contact, etc. I might limit it to just changes that are directly on the Opportunity, such as when the actual relationship field changes to a new value (which is the use case John defines) and not when the distant field changes if the record doesn't. Thanks! @mpusto
